src/main/java/com/epam/test_generator/services/JiraService.java

```diff
@@ -0,0 +1,332 @@
+                .collect(Collectors.toList());
+    }
+
+
+    /**
+     * Synchronisation method from Jira
+     * This method receives all related and opened stories and sub stories from Jira and update its in BDD.
+     * <p>
+     * In case where issue has closed in Jira - the changes not present in the response. By this reason
+     * all existing cases or suits which not presents in response will be deleted from BBD.
+     */
+    public void syncFromJiraToBDD() throws JiraException {
+        String projects = projectDAO.findAll().stream()
+                .filter(p -> p.getJiraKey() != null)
+                .map(Project::getJiraKey)
+                .collect(Collectors.joining(","));
+
+        String query = String.format(
+                "project in (%s) AND status in (Open, \"In Progress\", Reopened, Verified) AND type=story",
+                projects);
+
+        String subtaskQuery = String.format(
+                "project in (%s) AND status in (Open, \"In Progress\", Reopened, Verified) AND type=sub-task",
+                projects);
+
+        List<JiraSubTask> jiraSubTasks = jiraSubStoryDAO.getJiraSubtoriesByFilter(subtaskQuery);
+
+        Map<String, Suit> suits = suitDAO.findAll().stream()
+                .collect(Collectors.toMap(Suit::getJiraKey, suit -> suit));
+        Map<String, Case> cases = caseDAO.findAll().stream()
+                .collect(Collectors.toMap(Case::getJiraKey, caze -> caze));
+
+        List<JiraStory> jiraStories = jiraStoryDAO.getJiraStoriesByFilter(query);
+
+        jiraStories.forEach(jiraStory -> {
+            updateSuitFromJiraStory(jiraStory);
+            suits.remove(jiraStory.getJiraKey());
+        });
+
+        jiraSubTasks.stream()
+                .filter(subTask -> suitDAO.findByJiraKey(subTask.getJiraParentKey()) != null)
+                .forEach(subTask -> {
+                    Case aCase = caseDAO.findByJiraKey(subTask.getJiraKey());
+                    if (aCase == null) {
+                        createCaseFromJiraSubTask(subTask);
+                    } else {
+                        updateSubTaskFromJiraSubTask(subTask);
+                        cases.remove(subTask.getJiraKey());
+                    }
+                });
+
+        suits.forEach((jiraKey, suit) -> {
+            projectDAO.findByJiraKey(suit.getJiraProjectKey()).getSuits().remove(suit);
+            suitDAO.delete(suit);
+        });
+        cases.forEach((jiraKey, caze) -> {
+            suitDAO.findByJiraKey(caze.getJiraParentKey()).getCases().remove(caze);
+            caseDAO.delete(caze);
+        });
+
+    }
+
+
+    /**
+     * Synchronisation method to Jira
+     * The method puts all changes in BBD (all changed suits=stories, cases=sub stories) to Jira
+     * in case when suit or case has deleted - stories/sub stories will be closed in Jira too
+     */
+    public void syncToJiraFromBDD() throws JiraException {
```

Comment:
I think you can extract inner if condition and inner loop to another private methods.